gammapy/background/energy_offset_array.py

```diff
@@ -95,12 +100,101 @@ def plot_image(self, ax=None, offset=None, energy=None, **kwargs):
         ax.set_title('Energy_offset Array')
         ax.legend()
-        image = ax.imshow(self.data, extent=extent, **kwargs)
+        image = ax.imshow(self.data.value, extent=extent, **kwargs)
         plt.colorbar(image)
         return ax
+
+    @classmethod
+    def read(cls, filename):
+        """Create `EnergyOffsetArray` from  fits file.
+
+        Parameters
+        ----------
+        filename : str
+            File name
+        """
+        hdu_list = fits.open(filename)
+        hdu = hdu_list[1]
+        return cls.from_fits(hdu)
+
+    @classmethod
+    def from_fits(cls, hdu):
+        """Read `EnergyOffsetArray` from a fits binary table.
+
+        Parameters
+        ----------
+        hdu : `~astropy.io.fits.BinTableHDU`
+            HDU binary table for the EnergyOffsetArray.
+
+        Returns
+        -------
+        cube : `~gammapy.background.EnergyOffsetArray`
+            EnergyOffsetArray object.
+        """
+
+        header = hdu.header
+        data = hdu.data
+        # get offset and energy binning
+        offset_edges = _make_bin_edges_array(data['THETA_LO'].squeeze(), data['THETA_HI'].squeeze())
+        energy_edges = _make_bin_edges_array(data['ENERG_LO'].squeeze(), data['ENERG_HI'].squeeze())
+
+        # get data
+        energy_offset_array = data['EnergyOffsetArray'].squeeze()
+        return cls(energy_edges, offset_edges, energy_offset_array)
+
+    def to_table(self):
+        """Convert `EnergyOffsetArray` to astropy table format.
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table containing the EnergyOffsetArray.
+        """
+        # table
+        table = Table()
+        table['THETA_LO'] = [self.offset[:-1]]
+        table['THETA_HI'] = [self.offset[1:]]
+        table['ENERG_LO'] = [self.energy[:-1]]
+        table['ENERG_HI'] = [self.energy[1:]]
+        table['EnergyOffsetArray'] = [Quantity(self.data, " u ")]
+        table.meta['name'] = 'TO DEFINE'
+
+        return table
+
+    def to_fits(self):
+        """Convert `EnergyOffsetArray` to binary table fits format.
+
+        Returns
+        -------
+        tbhdu : `~astropy.io.fits.BinTableHDU`
+            Table containing the EnergyOffsetArray.
+        """
+        return table_to_fits_table(self.to_table())
+
+    def write(self, filename, **kwargs):
+        """ Write EnergyOffsetArray to fits file"""
+        self.to_fits().writeto(filename, **kwargs)
+
+    def evaluate(self, energy, offset):
+        """
+        Interpolate the value of the `EnergyOffsetArray` at a given offset and Energy
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+         energy value
+        offset : `~astropy.coordinates.Angle`
+        offset value
+
+        Returns
+        -------
+        Interpolated value
+        """
+        Interpolator = interpolate.interp2d(self.offset.value, self.energy.value, self.data.value, fill_value="None")
```

Comment:
This doesn't work properly, does it?

Start by adding a test where you create a test array with known values.
This is already non-trivial how to create a good example.
One option could be to write some code that fills `value = energy[TeV] * offset[deg]` or `value = log10(energy[TeV]) * offset[deg]`
Then call `evaluate` and see if you get (roughly) the right value out for one or two positions.

My preference would be that we use the `RegularGridInterpolator` class as much as possible and not `interp2d`.
Example: https://github.com/gammapy/gammapy/blob/ea69461865ab46a16b165faf61ae5ce75b2f1b47/gammapy/irf/energy_dispersion.py#L801